packages/backend/src/apps/lettersg/actions/create-letter/index.ts

```diff
@@ -90,7 +115,26 @@ const action: IRawAction = {
       const rawResponse = await $.http.post('/v1/letters', payload)
       const response = responseSchema.parse(rawResponse.data)
 
-      $.setActionItem({ raw: response })
+      if (
+        !$.step.parameters.shouldGeneratePdf ||
+        !$.flow.hasFileProcessingActions
+      ) {
+        $.setActionItem({
+          raw: response,
+        })
+        return
+      }
+
+      const attachmentS3Key = await downloadAndStoreAttachmentInS3(
+        $,
+        response.publicId,
+      )
+      $.setActionItem({
+        raw: {
+          ...response,
+          attachment: attachmentS3Key,
+        },
+      })
```

Comment:
If you agree with my nit [above](https://github.com/opengovsg/plumber/pull/464/files#r1521238011):

```suggestion
      $.setActionItem({
        raw: {
          responseData: {
            ...response,
            attachment: attachmentS3Key,
          }
        },
      })
```